hummingbot/client/ui/interface_utils.py

```diff
@@ -59,25 +59,28 @@ async def start_trade_monitor(trade_monitor):
-                    for market, symbol in market_info:
-                        cur_trades = [t for t in trades if t.market == market and t.symbol == symbol]
-                        cur_balances = await hb.get_current_balances(market)
-                        perf = await PerformanceMetrics.create(market, symbol, cur_trades, cur_balances)
-                        return_pcts.append(perf.return_pct)
-                        pnls.append(perf.total_pnl)
-                    avg_return = sum(return_pcts) / len(return_pcts) if len(return_pcts) > 0 else s_decimal_0
-                    quote_assets = set(t.symbol.split("-")[1] for t in trades)
-                    if len(quote_assets) == 1:
-                        total_pnls = f"{PerformanceMetrics.smart_round(sum(pnls))} {list(quote_assets)[0]}"
-                    else:
-                        total_pnls = "N/A"
-                    trade_monitor.log(f"Trades: {len(trades)}, Total P&L: {total_pnls}, Return %: {avg_return:.2%}")
-                    return_pcts.clear()
-                    pnls.clear()
-        await _sleep(2)  # sleeping for longer to manage resources
+        try:
+            if hb.strategy_task is not None and not hb.strategy_task.done():
+                if all(market.ready for market in hb.markets.values()):
+                    trades: List[TradeFill] = hb._get_trades_from_session(int(hb.init_time * 1e3),
+                                                                          config_file_path=hb.strategy_file_name)
+                    if len(trades) > 0:
+                        market_info: Set[Tuple[str, str]] = set((t.market, t.symbol) for t in trades)
+                        for market, symbol in market_info:
+                            cur_trades = [t for t in trades if t.market == market and t.symbol == symbol]
+                            cur_balances = await hb.get_current_balances(market)
+                            perf = await PerformanceMetrics.create(market, symbol, cur_trades, cur_balances)
+                            return_pcts.append(perf.return_pct)
+                            pnls.append(perf.total_pnl)
+                        avg_return = sum(return_pcts) / len(return_pcts) if len(return_pcts) > 0 else s_decimal_0
+                        quote_assets = set(t.symbol.split("-")[1] for t in trades)
+                        if len(quote_assets) == 1:
+                            total_pnls = f"{PerformanceMetrics.smart_round(sum(pnls))} {list(quote_assets)[0]}"
+                        else:
+                            total_pnls = "N/A"
+                        trade_monitor.log(f"Trades: {len(trades)}, Total P&L: {total_pnls}, Return %: {avg_return:.2%}")
+                        return_pcts.clear()
+                        pnls.clear()
+            await _sleep(2)  # sleeping for longer to manage resources
+        except Exception:
+            hb.logger().exception("start_trade_monitor failed.")
```

Comment:
I think it will be useful to add details of the exception in the log. We can add them inside parenthesis at the end of the current message.